cgo/proofs.go

```diff
@@ -0,0 +1,333 @@
+package cgo
+
+/*
+#cgo LDFLAGS: -L${SRCDIR}/..
+#cgo pkg-config: ${SRCDIR}/../filcrypto.pc
+#include "../filcrypto.h"
+#include <stdlib.h>
+*/
+import "C"
+
+func VerifySeal(registeredProof RegisteredSealProof, commR *ByteArray32, commD *ByteArray32, proverId *ByteArray32, ticket *ByteArray32, seed *ByteArray32, sectorId uint64, proof SliceRefUint8) (bool, error) {
+	resp := C.verify_seal(registeredProof, commR, commD, proverId, ticket, seed, C.uint64_t(sectorId), proof)
```

Comment:
We can do it in a followup, but we should be checking that, e.g., `RegisteredSealProof` is a valid value for the rust enum.

Reply:
our new ffi bindings generate checks for this now https://getditto.github.io/safer_ffi/ffi-export/sanity-checks.html

Reply:
Oh that's wonderful.